src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -0,0 +1,338 @@
+                _log.LogDebug($"Full resolved path:'{_absoluteAdbExePath}'");
+            }
+        }
+
+        #endregion
+
+        #region Functions
+
+        public string GetAdbVersion()
+        {
+            return RunAdbCommand("version").StandardOutput;
+        }
+
+        public string GetAdbState()
+        {
+            return RunAdbCommand("get-state").StandardOutput;
+        }
+
+        public void ClearAdbLog()
+        {
+            RunAdbCommand("logcat -c");
+        }
+
+        public void DumpAdbLog(string outputFilePath, string filterSpec = "")
+        {
+            // This is somehow reaching full timeout, have to figure it out, allow 20s for demo?
+            var result = RunAdbCommand($"logcat -d {filterSpec}");
+            if (result.ExitCode != 0)
+            {
+                // Could throw here, but it would tear down a possibly otherwise acceptable execution.
+                _log.LogError($"Error getting ADB log:{Environment.NewLine}{FormatProcessOutputs(result)}");
+            }
+            else
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(outputFilePath));
+                File.WriteAllText(outputFilePath, result.StandardOutput);
+                _log.LogInformation($"Wrote current ADB log to {outputFilePath}");
+            }
+        }
+
+        public void StartAdbServer()
+        {
+            var result = RunAdbCommand("start-server");
+            _log.LogDebug($"{result.StandardOutput}");
+            if (result.ExitCode != 0)
+            {
+                throw new Exception($"Error starting ADB Server.  Std out:{result.StandardOutput} Std. Err: {result.StandardError}");
+            }
+        }
+
+        public void KillAdbServer()
+        {
+            var result = RunAdbCommand("kill-server");
+            if (result.ExitCode != 0)
+            {
+                throw new Exception($"Error killing ADB Server.  Std out:{result.StandardOutput} Std. Err: {result.StandardError}");
+            }
+        }
+
+        public int InstallApk(string apkPath)
+        {
+            _log.LogInformation($"Attempting to install {apkPath}: ");
+            var result = RunAdbCommand($"install {apkPath}");
+            if (result.ExitCode != 0)
+            {
+                _log.LogError($"Error:{Environment.NewLine}{FormatProcessOutputs(result)}");
+            }
+            else
+            {
+                _log.LogInformation($"Successfully installed {apkPath}.");
+            }
+            return result.ExitCode;
+        }
+
+        public int UninstallApk(string apkName)
+        {
+            _log.LogInformation($"Attempting to remove apk '{apkName}': ");
+            var result = RunAdbCommand($"uninstall {apkName}");
+            if (result.ExitCode == 0)
+            {
+                _log.LogInformation($"Successfully uninstalled {apkName}.");
+            }
+            else if (result.ExitCode == 255)

```

Comment:
I prefer constants for this numbers if they are known. That way people not familiar with the code will understand what is going on.